### PR TITLE
Fix hidden block in front

### LINKED ISSUE #836: Block disable not is hidden
#### Symfony packages
sonata-project/admin-bundle              3.20.1 The missing Symfony Admin Generator
sonata-project/block-bundle              3.3.2  Symfony SonataBlockBundle
sonata-project/cache                     1.0.7  Cache library
sonata-project/cache-bundle              2.3.1  This bundle provides caching services
sonata-project/classification-bundle     3.3.1  Symfony SonataClassificationBundle
sonata-project/core-bundle               3.4.0  Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.2.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.1.5  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.2.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.7.1  Lightweight Exporter library
sonata-project/formatter-bundle          3.2.2  Symfony SonataFormatterBundle
sonata-project/google-authenticator      2.0.0  Library to integrate Google Authenticator into a PHP project
sonata-project/intl-bundle               2.3.1  Symfony SonataIntlBundle
sonata-project/media-bundle              3.5.1  Symfony SonataMediaBundle
sonata-project/notification-bundle       3.1.0  Symfony SonataNotificationBundle
sonata-project/page-bundle               3.4.1  This bundle provides a Site and Page management through container and block services
sonata-project/seo-bundle                2.2.0  Symfony SonataSeoBundle
sonata-project/timeline-bundle           3.1.1  Integrates SpyTimelineBundle into Sonata
sonata-project/translation-bundle        2.1.1  SonataTranslationBundle
sonata-project/user-bundle               3.2.4  Symfony SonataUserBundle

#### PHP version
PHP 7.0

## Subject

When disable block in admin panel. Block does not hidden in frontend.

## Steps to reproduce
Change BlockInteractor in line ~163 add line 
`if ($block->getEnabled() == false) continue;`

## Expected results
Hidden block in frontend.

### PR DESCRIPTION
**Pull request by Migo2468**
Hidden block what they have disabled in admin panel. 

Closes #836 

```markdown
### Fixed
- fix #836 work disabled block in frontend.
```

### CODE REVIEW

File: Entity/BlockInteractor.php [hunk 1]
```diff
@@ -160,6 +160,7 @@ public function loadPageBlocks(PageInterface $page)
         $page->disableBlockLazyLoading();
 
         foreach ($blocks as $block) {
+            if ($block->getEnabled() == false) continue;
```
Comment (greg0ire) [inline Entity/BlockInteractor.php]: - Please add braces
- Please do not compare booleans to booleans. You already have a boolean perfectly usable in your condition : `!$block->getEnabled()`

**Comment by Migo2468** [comment]
Now ok?

File: Entity/BlockInteractor.php [hunk 2]
```diff
@@ -160,6 +160,9 @@ public function loadPageBlocks(PageInterface $page)
         $page->disableBlockLazyLoading();
 
         foreach ($blocks as $block) {
+            if (!$block->getEnabled()) {
```
Comment (greg0ire) [inline Entity/BlockInteractor.php]: There is a trailing space at the end of the line, this is why StyleCI complains. Please remove it.

**Comment by greg0ire** [comment]
Please write a changelog directed at end users. Keep in mind that they probably do not care what file or class you changed, but what features you provided / what bugs you fixed. Try to answer the question "What benefit do I get from upgrading?"
More on http://keepachangelog.com

File: Entity/BlockInteractor.php [hunk 3]
```diff
@@ -160,6 +160,9 @@ public function loadPageBlocks(PageInterface $page)
...
+                continue;
```
Comment (greg0ire) [inline Entity/BlockInteractor.php]: Same here apparently

**Comment by OskarStark** [comment]
Is this already covered by a test here?
https://github.com/sonata-project/SonataPageBundle/blob/3.x/Tests/Model/BlockInteractorTest.php

**Comment by Migo2468** [comment]
Hi it is not all.
If user is login in editor, block is disabled. But when logout or open page in Incognito block is show.
I do not know what controller show this blocks when we are logout. 

**Some hint?**

File: Entity/BlockInteractor.php [hunk 4]
```diff
@@ -160,6 +160,9 @@ public function loadPageBlocks(PageInterface $page)
         $page->disableBlockLazyLoading();
 
         foreach ($blocks as $block) {
+            if (!$block->getEnabled()) {
```
Comment (core23) [inline Entity/BlockInteractor.php]: Maybe we should filter this in the query instead of fetching ALL blocks and filtering them afterwards

**Comment by core23** [comment]
@Migo2468 are you still working on this?

**Comment by core23** [comment]
This PR looks dead. Feel free to reopen if you will continue working on this.

**Comment by Firensis** [comment]
Hmm, I want to note there is no actually a bug here. Disabled blocks are hidden if parameter hide_disabled_blocks is set to true in sonata_page.yaml.

sonata_page:
&nbsp;&nbsp;&nbsp;&nbsp;hide_disabled_blocks: true

**Pull request closed without merging**